src/js/index.js

```diff
@@ -1,22 +1,54 @@
 import React from 'react';
-import { render } from 'react-dom';
-import { App } from './App.jsx';
 import getMuiTheme from 'material-ui/styles/getMuiTheme';
 import injectTapEventPlugin from 'react-tap-event-plugin';
 import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import { render } from 'react-dom';
+import { Router, browserHistory } from 'react-router';
+
+import AboutRoute from './routes/about';
+import All from './routes/all';
+import App from './components/App.jsx';
+import ApiRoute from './routes/api';
 import { globalStyles } from './globalStyles.js';
+import LegendRoute from './routes/legend';
+import ResultRoute from './routes/result';
+
+// Apply app wide styles here (http://www.material-ui.com/#/customization/themes)
+const muiTheme = getMuiTheme(globalStyles);
+
+// react-router routes main config
+const routesConfig = {
+  path: '/',
+  // required - always load
+  component: App,
+  // load when on index only, on-demand
+  getIndexRoute(partialNextState, callback) {
+    require.ensure([], (require) => {
+      callback(null, {
+        component: require('./components/Search.jsx').default,
+      });
+    });
+  },
+  // loaded later on-demand
+  // https://github.com/ReactTraining/react-router/blob/master/docs/guides/DynamicRouting.md#dynamic-routing
```

Comment:
Is this over kill for this particular project? Dynamic loading of content is nice, but as that page itself says, is intended for a big site like facebook with massive amounts of changing, dynamic content.

The implementation like shown in [this tutorial](https://scotch.io/tutorials/routing-react-apps-the-complete-guide#more-routes)video seems far easier to follow and probably all that is needed here. Also doesn't hammer require which seems very ES5. Also uses components for route content as I suggested in another comment. As I said there, React.PureComponent would probably be a good optimisation for components that are equivalent to static pages.